2020/03/index.js

```diff
@@ -1,3 +1,53 @@
+const { promises: fs } = require('fs');
+
+async function getInput() {
+  return input = (await fs.readFile(`${__dirname}/input.txt`))
+    .toString()
+    .trim()
+    .split('\n')
+    .map(line => line.split(''));
+}
+
+async function p1(input, iIncrement = 3, jIncrement = 1) {
+  const tree = '#';
+  input = input
+    ? await Promise.resolve(input)
+    : await getInput();
+
+  let i = 0;
+  let numTrees = 0;
+  for(let j = 0; j < input.length; j += jIncrement) {
+    const row = input[j];
+    if(row[i % row.length] === tree) {
+      numTrees += 1;
+    }
+
+    i += iIncrement;
+  }
+
+  return numTrees;
+}
+
+async function p2() {
+  const input = await getInput();
+  const [a, b, c, d, e] = await Promise.all([
```

Comment:
such promise heavy code! Is this because of the file reading, or is this how you write node nowadays?

Reply:
Oh because I reused `p1()`. Yea looks pretty bad the way I set it up (`p1()` being  async).